### PR TITLE
gh-124538: Fix crash when using `gc.get_referents` on an untracked capsule object

### PR DESCRIPTION
**Pull request by ZeroIntensity**
* Issue: gh-124538

### CODE REVIEW

**Review by JelleZijlstra — APPROVED** [review]
cc @vstinner who implemented this in #108417.

File: Lib/test/test_gc.py [hunk 1]
```diff
@@ -1048,6 +1048,24 @@ class Z:
         callback.assert_not_called()
         gc.enable()
 
+    @cpython_only
+    def test_get_referents_on_capsule(self):
+        # See GH-124538
```
Comment (vstinner) [inline Lib/test/test_gc.py]: Would you mind to elaborate?

**Review by vstinner — APPROVED** [review]
LGTM

File: Lib/test/test_gc.py (hunk 1 above)
Comment (vstinner) [inline Lib/test/test_gc.py]:
```suggestion
        # gh-124538: Calling gc.get_referents() on an untracked capsule must not crash.
```

**Comment by vstinner** [comment]
Merged, thanks.

**Pull request merged**